sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -0,0 +1,206 @@
+} dif_rstmgr_result_t;
+
+/**
+ * Creates DIF Reset Manager state structure.
+ *
+ * Information that must be retained, and is required to program Reset Manager,
+ * shall be stored in `rstmgr`.
+ *
+ * @param base_addr Base address of an instance of the PLIC IP block.
+ * @param rstmgr Reset Manager state data.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_create(mmio_region_t base_addr,
+                                      dif_rstmgr_t *rstmgr);
+
+/**
+ * Configures DIF Reset Manager.
+ *
+ * Configures DIF Reset Manager using information in `config`. When `reset`
+ * set to true, the peripheral is reset to sane defaults before configuring.
+ *
+ * @param rstmgr Reset Manager state data.
+ * @param config Reset Manager configuration data.
+ * @param reset Reset before configuring.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_configure(const dif_rstmgr_t *rstmgr,
+                                         const dif_rstmgr_config_t *config,
+                                         bool reset);
+
+/**
+ * Resets the Reset Manager.
+ *
+ * Resets all the Reset Manager registers.
+ *
+ * @param rstmgr Reset Manager state data.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_reset(const dif_rstmgr_t *rstmgr);
+
+/**
+ * Obtains the complete Reset Manager reset causes information.
+ *
+ * The reset causes are parsed and presented to the caller as an
+ * array of flags in 'causes'.
+ *
+ * @param rstmgr Reset Manager state data.
+ * @param causes Reset causes information.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_causes_get(const dif_rstmgr_t *rstmgr,
+                                          dif_rstmgr_causes_bitfield_t *causes);
+
+/**
+ * Clears the reset causes information in Reset Manager.
+ *
+ * @param rstmgr Reset Manager state data.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_causes_clear(const dif_rstmgr_t *rstmgr);
+
+/**
+ * `dif_rstmgr_software_reset` specific results.
+ */
+typedef enum dif_rstmgr_software_reset_result {
+  kDifRstmgrSoftwareResetOk = kDifRstmgrOk,
+  kDifRstmgrSoftwareResetError = kDifRstmgrError,
+  kDifRstmgrSoftwareResetBadArg = kDifRstmgrBadArg,
+  /**
+   * Software reset is locked for the requested peripheral.
+   */
+  kDifRstmgrSoftwareResetLocked,
+} dif_rstmgr_software_reset_result_t;
+
+/**
+ * Asserts or de-asserts software reset for the requested peripheral.
+ *
+ * @param rstmgr Reset Manager state data.
+ * @param peripheral Peripheral to assert/de-assert reset for.
+ * @param reset Reset control.
+ * @return `dif_rstmgr_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_software_reset_result_t dif_rstmgr_software_reset(
+    const dif_rstmgr_t *rstmgr, dif_rstmgr_peripheral_t peripheral,
+    dif_rstmgr_software_reset_t reset);
```

Comment:
Is there a way of reading the current `hold in reset` value, or is the register write-only?

Reply:
There is, I will add a routine to check it.